test/unit/Cover/constructor.js

```diff
@@ -0,0 +1,30 @@
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+
+describe('constructor', function () {
+  it('should set variables correctly', async function () {
+    const { quotationData, stakingPool, futureCoverNFTAddress, coverAddress, coverUtilsLib } = this;
+
+    const Cover = await ethers.getContractFactory('Cover', {
+      libraries: {
+        CoverUtilsLib: coverUtilsLib.address,
+      },
+    });
+
+    const cover = await Cover.deploy(
+      quotationData.address,
```

Comment:
these values

quotationData.address
productsV1.address

are private fields. However we can try using getStorageAt https://docs.ethers.io/v5/api/providers/provider/#Provider-getStorageAt

to get the field value.

Let's try asserting them regardless using getStorageAt

Reply:
I've tried with `hardhat-storage-layout` and couldn't find them, and then I realized that they are immutable so they are not saved to storage